### PR TITLE
Added ed25519 algo

### PR DESCRIPTION
**Pull request by micahhausler**
Added ed25519 algorithm and empty Digester

### CODE REVIEW

**Comment by micahhausler** [comment]
I need to actually test the empty Digester before I'll mark this review ready

File: contentdigest/digester_test.go [hunk 1]
```diff
@@ -79,6 +79,12 @@ func TestDigester_HashRequest(t *testing.T) {
 			req:      io.NopCloser(mockReader{Err: errors.New("read error")}),
 			wantErr:  true,
 		},
+		{
+			name:     "empty",
+			digester: Empty,
+			req:      io.NopCloser(bytes.NewBufferString(`{"hello": "world"}`)),
+			want:     `empty=::`,
```
Comment (micahhausler) [inline contentdigest/digester_test.go]: I think what we actually want is just an empty string. I'll modify the `HashRequest` to bypass empty hasher

File: alg_ed25519/alg.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+func (a Ed25519) Attributes() any {
+	return a.Attrs
+}
+
+func (a Ed25519) Type() string {
+	return Ed25519Alg
+}
+
+func (a Ed25519) ContentDigest() contentdigest.Digester {
+	return contentdigest.SHA512
```
Comment (micahhausler) [inline alg_ed25519/alg.go:35]: I just picked sha512, since its independent of the actual signing algorithm. Let me know if you want something different

**Review by chrnorm — APPROVED** [review]

File: alg_ed25519/alg.go (hunk 2 above)
Comment (chrnorm) [reply alg_ed25519/alg.go:35]: Looks good, this could be made customisable in future too with a field on the `Ed25519` struct. It seems like in this instance, coupling the `ContentDigest` to the `Ed25519` is a poor abstraction because we force you to specify it in the algorithm.

Comment (micahhausler) [reply alg_ed25519/alg.go:35]: Yea, it might be nice to decouple those
